tests/test-NFA.cpp

```diff
@@ -0,0 +1,195 @@
+#include <cstdint>
+#include <map>
+#include <queue>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <log_surgeon/Constants.hpp>
+#include <log_surgeon/finite_automata/RegexAST.hpp>
+#include <log_surgeon/finite_automata/RegexNFA.hpp>
+#include <log_surgeon/Schema.hpp>
+#include <log_surgeon/SchemaParser.hpp>
+
+using log_surgeon::cSizeOfByte;
+using log_surgeon::finite_automata::RegexNFAByteState;
+using log_surgeon::Schema;
+using log_surgeon::SchemaVarAST;
+using std::string;
+
+using ByteLexicalRule = log_surgeon::LexicalRule<RegexNFAByteState>;
+using ByteNFA = log_surgeon::finite_automata::RegexNFA<RegexNFAByteState>;
+using RegexASTCatByte = log_surgeon::finite_automata::RegexASTCat<RegexNFAByteState>;
+using RegexASTCaptureByte = log_surgeon::finite_automata::RegexASTCapture<RegexNFAByteState>;
+using RegexASTGroupByte = log_surgeon::finite_automata::RegexASTGroup<RegexNFAByteState>;
+using RegexASTLiteralByte = log_surgeon::finite_automata::RegexASTLiteral<RegexNFAByteState>;
+using RegexASTMultiplicationByte
+        = log_surgeon::finite_automata::RegexASTMultiplication<RegexNFAByteState>;
+using RegexASTOrByte = log_surgeon::finite_automata::RegexASTOr<RegexNFAByteState>;
+
+TEST_CASE("Test NFA", "[NFA]") {
+    Schema schema;
+    string const var_name = "capture";
+    string const var_schema = var_name + string(":")
+                              + string("Z|(A(?<letter>((?<letter1>(a)|(b))|(?<letter2>(c)|(d))))B(?"
+                                       "<containerID>\\d+)C)");
```

Comment:
```suggestion
    string const var_name{"capture"};
    string const var_schema{
            var_name + ":"
            + "Z|(A(?<letter>((?<letter1>(a)|(b))|(?<letter2>(c)|(d))))B(?"
              "<containerID>\\d+)C)"
    };
```
1. Use `{}` initialization whenever possible
2. No need to explicitly add `string{}`